Library/Homebrew/keg_relocate.rb

```diff
@@ -165,6 +165,49 @@ def replace_text_in_files(relocation, files: nil)
     changed_files
   end
 
+  def relocate_build_prefix(keg, old_prefix, new_prefix)
+    # Find binaries which match prefix strings.
+    string_matches = Set.new
+    keg.each_unique_file_matching(old_prefix) do |file|
+      string_matches << file
+    end
+
+    binary_string_matches = Set.new
+    keg.each_unique_binary_file do |file|
+      binary_string_matches << file if string_matches.include?(file)
+    end
+
+    # Only consider string matches which are binary files with null bytes, and remove any
+    # matches which are sharballs found by text_files.
+    binary_string_matches -= text_files
```

Comment:
Similarly, rather than removing these files after reading them: consider not reading them at all.

Reply:
Yes, I should just remove `text_files` from `string_matches` earlier.  Combined with the above change, that will mean a much higher proportion of the files being read by `each_unique_binary` are likely to be binary files.

Reply:
Done.